Objective-C/CBLCollection.mm

```diff
@@ -21,45 +21,139 @@
 #import "CBLIndexable.h"
 #import "CBLChangeListenerToken.h"
 #import "CBLCollectionChangeObservable.h"
+#import "CBLDatabase+Internal.h"
+#import "CBLStringBytes.h"
+#import "CBLScope.h"
+#import "CBLIndexConfiguration+Internal.h"
+#import "CBLStatus.h"
+#import "CBLScope+Internal.h"
+
+using namespace fleece;
 
 NSString* const kCBLDefaultCollectionName = @"_default";
 
-@implementation CBLCollection
+@implementation CBLCollection {
+    __weak CBLDatabase* _db;
```

Comment:
This doesn't need to fix in this PR. The question is that how do you handle the case the CBLDatabase* pointer is dangle from the case that the database is released but the user still keep the collection object? When that happen, anything that uses _db in CBLCollection class will cause the crash including [db safeBlock:]. This is something that needs to be handled.

Reply:
Noted that if CBLDatabase doesn't cache any CBLCollections, you could retain db in CBLCollection.